### ISSUE TITLE
Feature Request: Default Video Quality

### ISSUE THREAD

**Issue by raddacle** [yuliskov/SmartTubeNext#246]
I don't see an option in the settings to set the default quality for videos. Where if our default quality is 4k, and a video didn't offer 4k, it would just select 1080p. At the moment, the app is picking 720p even though videos have higher options.

**Comment by foreveryoungforty** [comment]
Just play a 4 k video, you go via HQ to video preset , and there you choose the best video quality setting your device/TV can handle and render. From that moment the best quality is chosen for each video. On my Sony 4K HDR (where I choose 4K 60 vp9 HDR as preset) it works perfect.
Hope this helps you.

**Comment by fawzib** [comment]
we should just have an option to set quality to highest. or something like auto - high (keeps trying the best quality) or just the regular auto

**Comment by memturbo1** [comment]
Yes I don't really understand the presets .For me they don't seem to work properly
when I change the preset I still get videos in lower quality than the preset even tho the manual option is there .

**Comment by fawzib** [comment]
we just need an option to set best quality and it will automatically choose
what's best for that video. for example my tv is 4k hdr. I would like the
best quality to be that but if a video of 1080p what will happen ? will it
lower it to 1080p or just choose randomly .

**Comment by foreveryoungforty** [comment]
As the settings/options has altered through the beta-versions, you best do the following.
After the installation of the app, you go to settings > video player > video preset : according to the capabilities of your Android TV , you there choose the best video quality your android tv can handle, as I wrote earlier I have a Sony 4K HDR , so I choose 4K 60 vp9 HDR as video set. Don't choose higher nor lower. Compare to a the setup of a new blu-ray player, also there you have to choose in the video/screen-settings the best your tv can handle.
From that moment the best quality will be chosen for each video you play, **depending of course on the highest source quality that video has on YouTube.** .  So don't expect a 720p to play 4K (the app does not upscale !!), but a real 4K will play 4K....
So after that initial setting, don't worry anymore about video quality and stay away from hq in the video itself : the app will always choose the best possible quality depending on source and never higher than your initial preset.
It works perfect over here.

**Comment by fawzib** [comment]
so it works as intended I suggest we rename it to best quality
